src/client/Microsoft.Identity.Client/Internal/Requests/SilentRequestHelper.cs

```diff
@@ -41,70 +52,104 @@ internal static async Task<MsalTokenResponse> RefreshAccessTokenAsync(MsalRefres
-            {
-                int jitter = s_random.Next(-Constants.DefaultJitterRangeInSeconds, Constants.DefaultJitterRangeInSeconds);
-                var refreshOnWithJitter = msalAccessTokenCacheItem.RefreshOn.Value + TimeSpan.FromSeconds(jitter);
-                return refreshOnWithJitter;
-            }
-
-            
-            return null;           
+        internal static bool NeedsRefresh(MsalAccessTokenCacheItem oldAccessToken)
+        {
+            return NeedsRefresh(oldAccessToken, out _);
+        }
+
+        internal static bool NeedsRefresh(MsalAccessTokenCacheItem oldAccessToken, out DateTimeOffset? refreshOnWithJitter)
+        {
+            refreshOnWithJitter = GetRefreshOnWithJitter(oldAccessToken);
+            if (refreshOnWithJitter.HasValue && refreshOnWithJitter.Value < DateTimeOffset.UtcNow)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Fire and forget the fetch action on a background thread.
+        /// Do not change to Task and do not await it.
+        /// </summary>
+        internal static void ProcessFetchInBackground(
+            MsalAccessTokenCacheItem oldAccessToken,
+            Func<Task<AuthenticationResult>> fetchAction,
+            ICoreLogger logger)
+        {
+            _ = Task.Run(async () =>
+            {
+                try
+                {
+                    await fetchAction().ConfigureAwait(false);
+                }
+                catch (MsalServiceException ex)
+                {
+                    string logMsg = $"Background fetch failed with MsalServiceException. Is AAD down? { ex.IsAadUnavailable()}";
+                    if (ex.StatusCode == 400)
+                    {
+                        logger.ErrorPiiWithPrefix(ex, logMsg);
+                    }
+                    else
+                    {
+                        logger.WarningPiiWithPrefix(ex, logMsg);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    string logMsg = $"Background fetch failed with exception.";
+                    logger.WarningPiiWithPrefix(ex, logMsg);
+                }
+            });
+        }
+
+        private static Random s_random = new Random();
+        private static DateTimeOffset? GetRefreshOnWithJitter(MsalAccessTokenCacheItem msalAccessTokenCacheItem)
+        {
+            if (msalAccessTokenCacheItem.RefreshOn.HasValue)
+            {
+                int jitter = s_random.Next(-Constants.DefaultJitterRangeInSeconds, Constants.DefaultJitterRangeInSeconds);
+                var refreshOnWithJitter = msalAccessTokenCacheItem.RefreshOn.Value + TimeSpan.FromSeconds(jitter);
+                return refreshOnWithJitter;
+            }
+
+            
+            return null;           
+        }
+
+        private static string ExtractEnrollmentId(string enrollmentIDs)
```

Comment:
I would put this into iOS proxy GetEntrollmentID method.

Reply:
Placed in #ifiOS block